awssdk/src/main/scala/Client.scala

```diff
@@ -170,6 +170,17 @@ trait Client[F[_]] {
     parallelism: Int
   ): fs2.Stream[F, U]
 
+  /** Scan a table. Allows to pass an initialKey from where to start
+   * scanning.
+   * Returns the stream with the lastKey. Useful to use for checkpointing.
+    */
+  def scan[U: Decoder](
+    tableName: String,
+    consistentRead: Boolean,
+    parallelism: Int,
+    initialKey: Option[java.util.Map[String, AttributeValue]] = None
```

Comment:
since this is overloading `scan` which doesn't have `initialKey` param, I think it is better to have this parameter explicit as an immutable scala `Map`. This allows client to stay purely Scala without creating java `Map`. 

For example:
```
def scan[U: Decoder](
    tableName: String,
    consistentRead: Boolean,
    parallelism: Int,
    startKey: Map[String, AttributeValue])
```

Reply:
That is done. I am happy to move the Scala <-> Java map conversion further in the `ScanOps` class if you think it is better.

While at it I noticed there is another `scan` method that takes a `filterExpression` param, and the implementation is almost the exact same as the other. Do you want me to refactor that method to be only one with the following signature? Maybe it is too much changes and you'd rather do it in following PRs (or it is just a sh* idea :sweat_smile: )

```
  def scan[U: Decoder](
    tableName: String,
    consistentRead: Boolean,
    maybeFilter: Option[Expression] = None,
    parallelism: Int,
    initialKey: Map[String, AttributeValue] = Map.empty
  ): fs2.Stream[F, (Map[String, AttributeValue], U)]
```

Reply:
great spot! there is a `Monoid[Expression]` so you don't need `Option[Expression]` btw. Please feel free to refactor the method.